services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/service/tenant/TenantInformationService.java

```diff
@@ -56,12 +56,12 @@
      *             The <em>status</em> will be
      *             <ul>
      *             <li><em>200 OK</em> if a tenant with the given ID is registered.
-     *             The <em>payload</em> will contain the tenant's configuration information.</li>
+     *             The <em>payload</em> will contain the tenant information.</li>
      *             <li><em>404 Not Found</em> if no tenant with the given identifier exists.</li>
      *             </ul>
      *
      * @throws NullPointerException if any argument is {@code null}.
      */
-     Future<TenantResult<TenantObject>> getTenant(String tenantId, Span span);
+     Future<OperationResult<Tenant>> getTenant(String tenantId, Span span);
```

Comment:
Is it really useful to return an `OperationResult` here? In the past it has merely caused confusion about how to interpret it correctly. FMPOV it is much more straightforward to either succeed the future with a *Tenant* or fail it with a *ServiceInvocationException* ...

Reply:
I have pushed changes for the above comment .  Also created a `DefaultTenantInformationService` without using `@Autowired` and the device registries are configured use this new class.